core/controllers/question_test.py

```diff
@@ -111,3 +111,350 @@ def test_get(self):
+        self.question = question_services.get_question_by_id(question_id)
+        question_services.add_question_id_to_skill(
+            self.question.question_id, self.question.collection_id,
+            self.skill_id, self.owner_id)
+
+        self.payload = {}
+
+    def test_get(self):
+        """Test to verify get method."""
+        self.payload['collection_id'] = self.collection_id
+        response_json = self.get_json(
+            '%s' % feconf.QUESTION_MANAGER_URL, self.payload,
+            expect_errors=False)
+        expected_question_summary = question_domain.QuestionSummary(
+            self.question.question_id, self.question.title, ['test'])
+        self.assertIn(
+            expected_question_summary.to_dict(),
+            response_json['question_summary_dicts'])
+
+        response = self.testapp.get(
+            '%s/batch' % feconf.QUESTION_MANAGER_URL,
+            expect_errors=True)
+        self.assertEqual(response.status_int, 404)
+
+        del self.payload['collection_id']
+        response = self.testapp.get(
+            '%s/batch' % feconf.QUESTION_MANAGER_URL, self.payload,
+            expect_errors=True)
+        self.assertEqual(response.status_int, 404)
+
+
+class QuestionIntegrationTest(test_utils.GenericTestBase):
+    """Test the handlers for managing questions."""
+
+    def setUp(self):
+        super(QuestionIntegrationTest, self).setUp()
+
+        self.collection_id = 'coll_0'
+        self.exp_id = 'exp_1'
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.new_user_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+
+        # Create a new collection and exploration.
+        self.save_new_valid_collection(
+            self.collection_id, self.new_user_id, exploration_id=self.exp_id)
+
+        # Add a skill.
+        collection_services.update_collection(
+            self.new_user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
+                'name': 'test'
+            }], 'Add a new skill')
+        collection = collection_services.get_collection_by_id(
+            self.collection_id)
+        self.skill_id = collection.get_skill_id_from_skill_name('test')
+        collection_node = collection.get_node(self.exp_id)
+        collection_node.update_acquired_skill_ids([self.skill_id])
+
+        # Update the acquired skill IDs for the exploration.
+        collection_services.update_collection(
+            self.new_user_id, self.collection_id, [{
+                'cmd': collection_domain.CMD_EDIT_COLLECTION_NODE_PROPERTY,
+                'property_name': (
+                    collection_domain.COLLECTION_NODE_PROPERTY_ACQUIRED_SKILL_IDS), # pylint: disable=line-too-long
+                'exploration_id': self.exp_id,
+                'new_value': [self.skill_id]
+            }], 'Update skill')
+
+        self.payload = {}
+
+
+    def test_black_box(self):
```

Comment:
No clue what this does from the name. Please explain more clearly, either using a docstring or inline comments or a better function name or something.

Reply:
updated.